src/ScriptCoverageManager.cc

```diff
@@ -127,31 +124,48 @@ bool ScriptCoverageManager::WriteStats()
 		return false;
 		}
 
-	for ( auto s : stmts )
+	for ( auto& s : stmts )
 		{
-		ODesc location_info;
-		s->GetLocationInfo()->Describe(&location_info);
 		ODesc desc_info;
 		s->Describe(&desc_info);
 		string desc(desc_info.Description());
-		canonicalize_desc cd{delim};
-		for_each(desc.begin(), desc.end(), cd);
-		pair<string, string> location_desc(location_info.Description(), desc);
-		if ( usage_map.find(location_desc) != usage_map.end() )
-			usage_map[location_desc] += s->GetAccessCount();
-		else
-			usage_map[location_desc] = s->GetAccessCount();
+		TrackUsage(s, desc_info.Description(), s->GetAccessCount());
 		}
 
-	map<pair<string, string>, uint64_t>::const_iterator it;
-	for ( auto& um : usage_map )
+	for ( auto& fi : func_instances )
```

Comment:
```suggestion
	for ( auto& [func, body] : func_instances )
```

And remove the two variables from the body.